src/plugin.validation.dns.azure/AzureOptionsFactory.cs

```diff
@@ -46,6 +68,7 @@ public override Task<AzureOptions> Default(Target target)
             var options = new AzureOptions
             {
                 UseMsi = az.AzureUseMsi,
+                AzureEnvironment = _arguments.TryGetRequiredArgument(nameof(az.AzureEnvironment), az.AzureEnvironment),
```

Comment:
This will break existing workflows because `TryGetRequiredArgument` call will throw an exception when the argument you're asking for has not been provided. This should be an optional arguments and most preferably also be case insensitive, e.g. `AzureUSGovernment` and `AzureUsGovernment` should mean the same thing.